common/flatpak-run.c

```diff
@@ -1622,13 +1622,30 @@ job_removed_cb (SystemdManager *manager,
     g_main_loop_quit (data->main_loop);
 }
 
+static GString *
+systemd_unit_name_escape (const gchar *in)
+{
+  /* Adapted from systemd source */
+  GString * const ret = g_string_sized_new (strlen (in) * 4);
+
+  for (; *in; in++)
+    {
+      if (g_ascii_isalnum (*in) || *in == ':' || *in == '_' || *in == '.')
+        g_string_append_c (ret, *in);
+      else
+        g_string_append_printf (ret, "\\x%02x", *in);
+    }
+  return ret;
```

Comment:
When using GString to build strings you use a local GString and then you end with `return g_string_free (s, FALSE)` which returns the built up `char *` and frees the rest.

Reply:
the calling function stores it in a `g_autoptr` which frees everything, right ?